rust/src/wow/db.rs

```diff
@@ -128,23 +127,34 @@ impl Wdc4Db2File {
     }
 }
 
-fn bitslice_to_u32(bits: &BitSlice<u8, Msb0>, bit_offset: usize, num_bits: usize) -> u32 {
-    let mut result: u32 = 0;
-    for bit_num in bit_offset..bit_offset + num_bits {
-        let byte_index = bit_num >> 3;
-        let bit_index = 7 - (bit_num % 8);
-        if bits[byte_index * 8 + bit_index] {
-            result |= 1 << (bit_num - bit_offset);
+fn from_u32<T>(v: u32) -> Result<T, DekuError>
+    where for<'a> T: DekuReader<'a, ()>
+{
+    let mut v_bytes = v.to_le_bytes();
+    let mut cursor = Cursor::new(&mut v_bytes);
+    let mut reader = Reader::new(&mut cursor);
+    T::from_reader_with_ctx(&mut reader, ())
+}
+
+fn read_field_to_u32<R: deku::no_std_io::Read + deku::no_std_io::Seek>(reader: &mut Reader<R>, field_offset_bits: usize, field_size_bits: usize) -> Result<u32, DekuError> {
+    // Assumes the reader points to the start of the record
+    let old = reader.seek(std::io::SeekFrom::Current(0i64)).unwrap();
```

Comment:
[looks like they implemented `From<Error> for DekuError`](https://docs.rs/deku/latest/deku/error/enum.DekuError.html#impl-From%3CDekuError%3E-for-Error), so i think you should be able to just do:

```suggestion
    let old = reader.seek(std::io::SeekFrom::Current(0))?;
```

Reply:
Do I need to import a trait or something?

```
error[E0277]: `?` couldn't convert the error to `deku::DekuError`
   --> src\wow\db.rs:215:61
    |
215 |         let old = reader.seek(std::io::SeekFrom::Current(0))?;
    |                                                             ^ the trait `From<std::io::Error>` is not implemented for `deku::DekuError`, which is required by `std::result::Result<String, deku::DekuError>: FromResidual<std::result::Result<Infallible, std::io::Error>>`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following other types implement trait `From<T>`:
              <deku::DekuError as From<Infallible>>
              <deku::DekuError as From<TryFromSliceError>>
              <deku::DekuError as From<std::num::TryFromIntError>>
    = note: required for `std::result::Result<String, deku::DekuError>` to implement `FromResidual<std::result::Result<Infallible, std::io::Error>>` 
```

Reply:
oh i'm dumb, they implemented `From<DekuError> for Error` lol. looks like one thing you could do is

```
    let old = reader.seek(std::io::SeekFrom::Current(0))
        .map_err(|err| DekuError::Io(error.kind()))?;
```